pkg/scheduler/api/podgroup_info/job_info.go

```diff
@@ -182,31 +192,35 @@ func (podGroupInfo *PodGroupInfo) UpdateTaskStatus(task *pod_info.PodInfo, statu
 func (podGroupInfo *PodGroupInfo) deleteTaskIndex(ti *pod_info.PodInfo) {
 	if tasks, found := podGroupInfo.PodStatusIndex[ti.Status]; found {
 		delete(tasks, ti.UID)
+		if pod_status.IsActiveAllocatedStatus(ti.Status) {
+			podGroupInfo.ActiveAllocatedCount -= 1
+		}
 
 		if len(tasks) == 0 {
 			delete(podGroupInfo.PodStatusIndex, ti.Status)
 		}
+
+		podGroupInfo.invalidateTasksCache()
 	}
 }
 
-func (podGroupInfo *PodGroupInfo) GetActiveAllocatedTasks() []*pod_info.PodInfo {
-	var tasksToAllocate []*pod_info.PodInfo
-	for _, task := range podGroupInfo.PodInfos {
-		if pod_status.IsActiveAllocatedStatus(task.Status) {
-			tasksToAllocate = append(tasksToAllocate, task)
-		}
-	}
-	return tasksToAllocate
+func (podGroupInfo *PodGroupInfo) invalidateTasksCache() {
+	podGroupInfo.tasksToAllocate = nil
+	podGroupInfo.tasksToAllocateInitResource = nil
+}
+
+func (podGroupInfo *PodGroupInfo) GetActiveAllocatedTasksCount() int {
+	return podGroupInfo.ActiveAllocatedCount
 }
 
-func (podGroupInfo *PodGroupInfo) GetActivelyRunningTasks() []*pod_info.PodInfo {
-	var tasks []*pod_info.PodInfo
+func (podGroupInfo *PodGroupInfo) GetActivelyRunningTasksCount() int32 {
```

Comment:
Why specifically int32?

Reply:
The only call for this function have been casting it to int32. In that case, why not return an int32 in the first place?